supporter-product-data/src/main/scala/com/gu/services/SelectActiveRatePlansQuery.scala

```diff
@@ -0,0 +1,39 @@
+package com.gu.services
+
+import com.gu.model.FieldsToExport._
+import com.gu.model.Stage
+import com.gu.model.Stage.{DEV, PROD}
+
+import java.time.LocalDate
+
+object SelectActiveRatePlansQuery {
+
+  val name = "select-active-rate-plans"
+
+  val isNotDSGift = "Subscription.RedemptionCode__c = ''"
+  val isRedeemedDSGift = s"(Subscription.RedemptionCode__c != '' AND ${gifteeIdentityId.zuoraName} != '')"
+
+  def excludeDiscountProductRatePlans(discountProductRatePlanIds: List[String]) =
+    discountProductRatePlanIds
+      .map(discountProductRatePlanId => s"${productRatePlanId.zuoraName} != '$discountProductRatePlanId' AND")
+      .mkString("\n")
+
+  def query(date: LocalDate, discountProductRatePlanIds: List[String]): String =
+    s"""SELECT
+          ${identityId.zuoraName},
+          ${gifteeIdentityId.zuoraName},
+          ${ratePlanId.zuoraName},
+          ${productRatePlanId.zuoraName},
+          ${productRatePlanName.zuoraName},
+          ${termEndDate.zuoraName}
```

Comment:
traditionally the subscription validity is based on being covered by a charge rather than just being within term.  I think that was more of a concern when things were originally migrated in, and newly acquirable products have the charges set up to fill the subscription term.  Is this logic matching the old members-data-api logic?

Reply:
The query name `select-active-rate-plans` might be confusing as this query will not in all cases identify only the **active** rate plans, but also past ones and the ones far in the future (if there were multiple renewals amendments scheduled for the future). However if the intention of the query is to identify active subscriptions (as opposed to rateplans) it might be good enough, but perhaps this should be reflected in the scala object and zuora query names.